nx-X11/programs/Xserver/composite/compext.c

```diff
@@ -27,18 +27,44 @@
 #endif
 
 #include "compint.h"
+#include "XI.h"
+#include "XIproto.h"
+#include "extinit.h"
+
+#ifndef SERVER_COMPOSITE_MAJOR_VERSION
+#define SERVER_COMPOSITE_MAJOR_VERSION 0
+#endif
+
+#ifndef SERVER_COMPOSITE_MINOR_VERSION
+#define SERVER_COMPOSITE_MINOR_VERSION 4
+#endif
 
 static CARD8	CompositeReqCode;
-int		CompositeClientPrivateIndex;
+
+#ifndef NXAGENT_SERVER
```

Comment:
True, but it's not just the missing privates header, there's also stuff like Xace and generally a reworked Dix Permissions system (or the like.)

I used NXAGENT_SERVER, because when the randr extension was updated within nx-libs, the same macro was used to distinguish between the "new upstream implementation" and the "backported implementation", see for instance nx-X11/programs/Xserver/randr/randr.c:77.

I just went through with that whenever I needed to backport stuff that was incompatible with our old X.Org server base code.